Sources/VLCMediaList.m

```diff
@@ -315,14 +311,17 @@ - (void)mediaListItemAdded:(NSArray *)arrayOfArgs
     [self didChange:NSKeyValueChangeInsertion valuesAtIndexes:[NSIndexSet indexSetWithIndexesInRange:range] forKey:@"media"];
 }
 
-- (void)mediaListItemRemoved:(NSNumber *)index
+- (void)mediaListItemRemoved:(NSArray *)arguments
 {
-    [self willChange:NSKeyValueChangeRemoval valuesAtIndexes:[NSIndexSet indexSetWithIndex:[index intValue]] forKey:@"media"];
+    NSNumber *index = arguments[0][@"index"];
+    VLCMedia *deleted = arguments[0][@"media"];
+    
     dispatch_sync(_serialMediaObjectsQueue, ^{
-        [_mediaObjects removeObjectAtIndex:[index intValue]];
+        if ([_mediaObjects containsObject:deleted]) {
+            [_mediaObjects removeObject:deleted];
+        }
     });
-    [self didChange:NSKeyValueChangeRemoval valuesAtIndexes:[NSIndexSet indexSetWithIndex:[index intValue]] forKey:@"media"];
-
+    
```

Comment:
Why do you remove support for key-value coding here?

Reply:
Ah I forgot to ask you about this. I personally didn't know you could do this until I saw the code for it. So, is this a feature that is used by users when they already have Notifications and delegates they can check?

Reply:
This is a used feature throughout multiple places of VLCKit, notably VLCMedia and VLCMediaPlayer. Especially the VLC-iOS app uses it.